testsuite/src/test_libblis.h

```diff
@@ -54,6 +54,9 @@
 
 // For pthreads API.
 #include <pthread.h>
+#ifdef __APPLE__
```

Comment:
How about checking for `_POSIX_BARRIERS`? See http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/unistd.h.html

Reply:
I like this idea better than branching on `__APPLE__`, even if the two are functionally equivalent for now. @devinamatthews ?

Reply:
`_POSIX_BARRIERS`

Reply:
@kali Would you mind following Isuru's recommendation and changing your cpp guard to one that uses `_POSIX_BARRIERS`?

Given that the document states, "This symbol shall always be set to the value 200809L," I assume you can change line 57 above to
```c
#ifdef _POSIX_BARRIERS
```

Reply:
well, it's actually defined at -1 on macos...

Reply:
@kali BTW, thanks for pointing out the -1 behavior. It wasn't clear from the header webpage that -1 was a possible value given that the description claimed it was always set to 200809L.

Reply:
> @kali BTW, thanks for pointing out the -1 behavior. It wasn't clear from the header webpage that -1 was a possible value given that the description claimed it was always set to 200809L.

This is probably Apple being rebellious wherever they can. Care to comment, @devinamatthews, since you have more experience with standards?

Reply:
If you believe in the absolute truth of POSIX preprocessor symbols, then do the following:
```
#if defined(_POSIX_BARRIERS) && (_POSIX_BARRIERS == 200809L)
```
On the other hand, if you have ever experienced a platform that does not correctly define preprocessor symbol or has bugs in system libraries, then you you will write a configure test to determine that the implementation actually works, or at least compiles and links, and use a macro like `BLIS_HAS_PTHREAD_BARRIER`.

Reply:
At the risk of shaking anyone's faith in standards, see https://wolandscat.net/2018/06/22/standards-paper-as-toilet-paper-torvalds/.

Reply:
Thanks for your comments, Jeff.

I agree that the "right" way is to write a configure test to determine if it actually works. However, I don't have the patience for that right now, and anything I don't have the patience for I don't usually expect of others (though I am delighted when they go above and beyond). :)

Reply:
If it helps to understand my cynicism 

From page 40 of the Red Book "IBM System Blue Gene Solution: Blue Gene/P Application Development":
> 4.4 Shared memory support
> 
> Shared memory is supported in Dual mode and Virtual Node Mode. Shared memory in SMP mode is not necessary because each processor already has access to all of the node’s memory. Shared memory is allocated using standard Linux methods (shm_open() and mmap()).

Translation: application that use POSIX shared-memory without taking this into account will crash because `shm_open` does a check for SMP mode and returns without doing what it is supposed to, and no preprocessor macro will tell you this.

I can't remember if the documentation above was added before or after I had to root cause this bug in a Quantum Monte Carlo code by reading the operating system source code for a day.